portal-frontend/src/app/features/edit-application/edit-application.component.ts

```diff
@@ -71,26 +70,38 @@ export class EditApplicationComponent implements OnInit {
   }
 
   async onSaveAndExit() {
-    const localGovernmentName = this.localGovernment.getRawValue();
-    const localGovernmentUuid = this.localGovernments.find((lg) => lg.name == localGovernmentName);
+    if (this.firstForm.dirty) {
+      let localGovernment;
+      const localGovernmentName = this.localGovernment.getRawValue();
+      if (localGovernmentName) {
+        localGovernment = this.localGovernments.find((lg) => lg.name == localGovernmentName);
 
-    if (!localGovernmentUuid) {
-      this.localGovernment.setErrors({
-        invalid: true,
+        if (!localGovernment) {
+          this.localGovernment.setErrors({
+            invalid: true,
+          });
+          return;
+        }
+      }
+
+      await this.applicationService.updatePending(this.fileId, {
+        applicant: this.applicantName.getRawValue(),
+        localGovernmentUuid: localGovernment ? localGovernment.uuid : undefined,
       });
-      return;
     }
-
-    await this.applicationService.updatePending(this.fileId, {
-      applicant: this.applicantName.getRawValue(),
-      localGovernmentUuid: localGovernmentUuid!.uuid,
-      documents: this.files.map((file) => file.file),
-    });
     await this.router.navigateByUrl('/home');
   }
 
   async onSubmit() {
-    //RUN FORM VALIDATION
+    this.applicantName.setValidators([Validators.required]);
+    this.applicantName.updateValueAndValidity();
+
+    this.localGovernment.setValidators([Validators.required]);
+    this.localGovernment.updateValueAndValidity();
+
+    if (this.firstForm.valid) {
+      alert('App Submitted');
```

Comment:
is it for debugging?

Reply:
Kind of, this will eventually do something else but I wanted something to show when it passed and alert was the easiest thing.